sql/sql_expression.h

```diff
@@ -1301,14 +1301,24 @@ PREDECLARE_VALUE_DESCRIPTION(TupleExpression);
 */
 struct GenerateRowsWhereFunction {
 
+    enum Complexity {        
+       
+        CONSTANT = 0,
+        BETTER_THAN_TABLESCAN,        
+        UNFILTERED_TABLESCAN,
```

Comment:
When you provide no WHERE clause, or some WHERE clause that evaluate to always true. Basically anytime we want all rows.